### PR TITLE
utillib: add git info to version line (branch, commit hash)

### LINKED ISSUE #1097: add git commit hash to console output
Add the latest commit hash such that...

```
Powered by QBlocks (GHC-TrueBlocks 0.5.5-alpha) [a40af3a]
```

### PR DESCRIPTION
**Pull request by wildmolasses**
Resolves #1097

### CODE REVIEW

File: src/libs/utillib/options_base.cpp [hunk 1]
```diff
@@ -405,7 +405,7 @@ namespace qblocks {
         os << notes();
         if (!isReadme) {
             os << bBlue << "  Powered by QBlocks";
-            os << (isTestMode() ? "" : " (" + getVersionStr(true) + ")") << "\n" << cOff;
+            os << (isTestMode() ? "" : " (" + getVersionStr(true) + ")") << " [" << GIT_BRANCH << "@" << GIT_COMMIT_HASH << "]\n" << cOff;
```
Comment (tjayrush) [inline src/libs/utillib/options_base.cpp]: This is great, but do me a favor? Can you (1) enter the command `getBlock --latest`. You should see the Parity version string:

```
Parity-Ethereum//v2.3.4-beta-0e95db11d-20190220/x86_64-macos/rustc1.32.0
```

Let's make our string look like this up to but NOT including the operating system. So,

```
GHC-TrueBlocks//v2.3.4-alpha-COMMIT-YYYYMMDD
```

C++ has built-in __YEAR__ defines I think.

Also, did you think about putting this inside of getVersionStr? In that way, the change would take place everywhere. I think that would be okay.

Comment (tjayrush) [reply src/libs/utillib/options_base.cpp]: Done. #47.

Comment (wildmolasses) [reply src/libs/utillib/options_base.cpp]: I might like to include the branch name too, but I'll defer. The branch name can sort of be inferred by the commit hash.

Comment (wildmolasses) [reply src/libs/utillib/options_base.cpp]: Do you want the date the user built the software here, or the date of the commit?

**Comment by tjayrush** [comment]
Hmmmm…

Commit, I think.

If we ever produce compiled binaries, I would think that a user who builds the same code should get identical version strings.

-------------------------------
Thomas Jay Rush
TrueBlocks.io
jrush@trueblocks.io
(610) 937-2593

**Comment by wildmolasses** [comment]
Ok, should we go off UTC timestamp then?

**Comment by tjayrush** [comment]
Always UTC. As if there were nothing else. Useless complication otherwise. Thanks.

-------------------------------
Thomas Jay Rush
TrueBlocks.io
jrush@trueblocks.io
(610) 937-2593

**Comment by wildmolasses** [comment]
Included YYYYMMDD in version string. Not 100% sure I've got the timestamp locale correct.

**Pull request merged**